R/filter2.R

```diff
@@ -45,10 +45,21 @@ filter2 = function(x, filter, boundary = c("circular", "replicate")) {
          },
          ## pad with a given value
          linear = {
-           dx[1:2] = dx[1:2] + cf[1:2]
-           xpad = array(val, dx)
-           ## is there a better way of doing this?
-           x = do.call("[<-", c(quote(xpad), lapply(d, function(x) enquote(1:x)), quote(x)) )
+           x = imageData(x)
+           
+           dx[1:2] = dx[1:2] + df[1:2] - 1L
```

Comment:
In the special case when we extend the canvas by filling it uniformly with a given value, there is no need to extend it from both sides. Please revert to `dx[1:2] = dx[1:2] + cf[1:2]` which creates a re-usable border shared in the left/right and top/bottom directions.